chore(wallet)!: request data from market providers by token key instead of symbol

saledjenic opened the pull request:
What's changed:
- coingecko provider updated to use token key (chain+address) instead of symbol
- `marketManager` works with coingecko provider only (for now, until we add coingecko to the proxy), `cryptoCompare` and
`cryptoCompareProxy` providers have been removed as an option.
- cryptocompare providers' public functions marked as deprecated

Changed endpoints:
- `wallet_fetchPrices` accepts a list of token keys (chain-address pair for each token) instead a list of symbols and returns prices per token key
- `wallet_fetchAllCurrencyFormats` returns currency formats where each format type is identified by `tokenKey` instead `symbol`
- `wallet_getCachedCurrencyFormats` returns currency formats where each format type is identified by `tokenKey` instead `symbol`

Changed signals:
- `UpdatedPrices` of `wallet.suggested.routes` signal returns price/s per token key

saledjenic committed:
chore(wallet)_: remove tokens with an empty address from the fetched tokens list

saledjenic committed:
chore(coingecko)_: request data from coingecko provider by token key instead of symbol

Up until now the data were requested by coingecko using the symbol, but since there can be more
tokens with the same symbol, the only correct way is to use chain+token-address pair and map it
to coingecko ids (which are unique) and reverse the process when the data are retrieved.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (16)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 53b29cd3 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6486/1/) | 2025-03-28 11:19:34 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250328-111634-53b29c-pr6486.aar) |
| :heavy_check_mark: | 53b29cd3 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6486/1/) | 2025-03-28 11:19:37 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250328-111635-53b29c-pr6486.zip) |
| :heavy_check_mark: | 53b29cd3 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6486/1/) | 2025-03-28 11:21:12 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250328-111639-53b29c-pr6486.zip) |
| :heavy_check_mark: | 53b29cd3 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6486/1/) | 2025-03-28 11:21:55 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250328-111636-53b29c-pr6486.zip) |
| :heavy_check_mark: | 53b29cd3 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6486/1/) | 2025-03-28 11:22:07 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250328-111639-53b29c-pr6486.zip) |
| :heavy_check_mark: | 53b29cd3 | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6486/1/) | 2025-03-28 11:23:10 | ~6 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250328-111644-53b29c-pr6486.zip) |
| :heavy_check_mark: | 53b29cd3 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6486/1/) | 2025-03-28 11:25:35 | ~8 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6486/1//consoleText) |
| :heavy_check_mark: | 53b29cd3 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6486/1/) | 2025-03-28 11:53:56 | ~37 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6486/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | ccb6c706 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6486/2/) | 2025-03-28 11:39:19 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250328-113626-ccb6c7-pr6486.zip) |
| :heavy_check_mark: | ccb6c706 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6486/2/) | 2025-03-28 11:39:29 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250328-113624-ccb6c7-pr6486.aar) |
| :heavy_check_mark: | ccb6c706 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6486/2/) | 2025-03-28 11:40:51 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250328-113633-ccb6c7-pr6486.zip) |
| :heavy_check_mark: | ccb6c706 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6486/2/) | 2025-03-28 11:41:01 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250328-113629-ccb6c7-pr6486.zip) |
| :heavy_check_mark: | ccb6c706 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6486/2/) | 2025-03-28 11:41:53 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250328-113630-ccb6c7-pr6486.zip) |
| :heavy_check_mark: | ccb6c706 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6486/2/) | 2025-03-28 11:42:22 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250328-113627-ccb6c7-pr6486.zip) |
| :heavy_check_mark: | ccb6c706 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6486/2/) | 2025-03-28 11:43:47 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6486/2//consoleText) |
| :heavy_check_mark: | ccb6c706 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6486/2/) | 2025-03-28 12:29:30 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6486/2//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 046daf90 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6486/3/) | 2025-03-31 08:43:16 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250331-084019-046daf-pr6486.aar) |
| :heavy_check_mark: | 046daf90 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6486/3/) | 2025-03-31 08:43:27 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250331-084019-046daf-pr6486.zip) |
| :heavy_check_mark: | 046daf90 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6486/3/) | 2025-03-31 08:45:00 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250331-084023-046daf-pr6486.zip) |
| :heavy_check_mark: | 046daf90 | [#3](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6486/3/) | 2025-03-31 08:45:16 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250331-084029-046daf-pr6486.zip) |
| :heavy_check_mark: | 046daf90 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6486/3/) | 2025-03-31 08:45:44 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250331-084021-046daf-pr6486.zip) |
| :heavy_check_mark: | 046daf90 | [#3](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6486/3/) | 2025-03-31 08:46:47 | ~6 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250331-084024-046daf-pr6486.zip) |
| :heavy_multiplication_x: | 046daf90 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6486/3/) | 2025-03-31 08:48:22 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6486/3//consoleText) |
| :heavy_check_mark: | 046daf90 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6486/3/) | 2025-03-31 09:18:06 | ~37 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6486/3//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 2b864be1 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-6486/4/) | 2025-03-31 08:50:01 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250331-084659-2b864b-pr6486.zip) |
| :heavy_check_mark: | 2b864be1 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-6486/4/) | 2025-03-31 08:50:12 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250331-084658-2b864b-pr6486.aar) |
| :heavy_check_mark: | 2b864be1 | [#4](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-6486/4/) | 2025-03-31 08:51:30 | ~4 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250331-084707-2b864b-pr6486.zip) |
| :heavy_check_mark: | 2b864be1 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-6486/4/) | 2025-03-31 08:51:42 | ~4 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250331-084703-2b864b-pr6486.zip) |
| :heavy_check_mark: | 2b864be1 | [#4](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-6486/4/) | 2025-03-31 08:52:20 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250331-084703-2b864b-pr6486.zip) |
| :heavy_check_mark: | 2b864be1 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-6486/4/) | 2025-03-31 08:52:27 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250331-084700-2b864b-pr6486.zip) |
| :heavy_multiplication_x: | 2b864be1 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6486/4/) | 2025-03-31 08:55:54 | ~7 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-6486/4//consoleText) |
| :heavy_check_mark: | 2b864be1 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6486/4/) | 2025-03-31 09:53:37 | ~35 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-6486/4//consoleText) |

saledjenic committed:
chore(wallet)!: code updated to use token key instead of symbol

What's changed:
- `marketManager` works with coingecko provider only (for now, until we add coingecko to the proxy), `cryptoCompare` and
`cryptoCompareProxy` providers are removed as options.
- cryptocompare providers's public functions marked as deprecated

Changed endpoints:
- `wallet_fetchPrices` accepts list of token keys (chain-address pair for each token) instead list of symbols and returns prices per token key
- `wallet_fetchAllCurrencyFormats` returns currency formats where each format type is identified by `tokenKey` instead `symbol`
- `wallet_getCachedCurrencyFormats` returns currency formats where each format type is identified by `tokenKey` instead `symbol`

Changed signals:
- `UpdatedPrices` of `wallet.suggested.routes` signal returns price/s per token key

saledjenic commented:
Both, mobile and desktop clients have to be adapted to these changes before merging.

@smohamedjavid I know that you're working on adding some new chains, please let me know if you merge that before merging this PR, cause some updates will be needed either on your or my side, depending on who merges first.

@shivekkhurana who will work on the mobile app side to address these changes?

saledjenic commented:
This PR handles tokens per token key, where the token key represents `chainId-address` pair, which is the only correct way. 
But since we cannot do the aggregation per name/symbol in that case, we've decided to use the coin gecko token list, which maintains the unique ID across chains. Because of that, this PR (which has a similar approach to the things done here, in case we decide later to switch to this) https://github.com/status-im/status-go/pull/6497 was created, and that's the reason for closing this PR, cause it is not relevant at this moment.
